start-here.mdx

```diff
@@ -1,8 +1,153 @@
+
+If something goes wrong, a non-zero [exit code](/tvm/exit-codes) might be returned, no changes to state or code are saved, and no further messages are sent.
+
+![Trace overview](/resources/images/trace.png)
+
+The only reason any code can be executed is when some account has received a message. Internal messages can only be sent by another contract executing some code, and that contract must have received a message from somewhere too. In the end, every message can be considered part of some _trace_: a tree of messages between accounts that starts with an incoming external message, continues with internal messages, and possibly ends with some outgoing external messages.
+
+If an external service needs to extract data from the blockchain, it can call a [get method](/tvm/get-method): a handler of incoming external messages in the code of some account. It's called a "method" because it acts like one: the message provides its parameters, and its execution returns a value. Get methods cannot make any changes to the blockchain.
+
+The other source is a [wallet](/standard/wallets/how-it-works). A wallet is an account with a specific kind of smart contract deployed on it, which can handle _transfer_ incoming external messages: requests to send an internal message to another account. There are several [implementations](/standard/wallets/comparison) of wallets, with varying extra functionality and protection. [`V5R1`](/standard/wallets/v5) is the latest official general-purpose wallet. The text below describes only the functionality common to most wallets.
+
+A transfer message consists of a destination address and, optionally, an internal message, both serialized and signed with a private key. The wallet stores a public key and uses it to check that a transfer message was signed with the corresponding private key. If the check succeeds, it sends that internal message to the destination address. This ensures only the user (or a service) who knows the private key can use the wallet.
+
+Private and public [keys](https://en.wikipedia.org/wiki/Public-key_cryptography) are generated in a program or service outside the blockchain and determine the address of the wallet account. Their text representation is called a [mnemonic](/standard/wallets/mnemonics).
+
+Before the code of the wallet can be deployed on an account with an incoming external message, some other account has to transfer Toncoin to the wallet account. External messages cannot have any Toncoin attached to them, and otherwise there wouldn't be enough funds to pay fees to handle the message that deploys a wallet. This transfer to a new wallet account usually comes from an exchange or another user. In testnet, there is [a bot](/ecosystem/wallet-apps/web#add-funds-into-the-wallet) that sends test coins to an account for free.
```

Comment:
"External messages cannot ..., and otherwise ..."

I think you supposed the "otherwise" part to be related to the previous sentence, but it's not clear from the way it's written right now